sdks/java/core/src/main/java/org/apache/beam/sdk/util/MutationDetectors.java

```diff
@@ -148,13 +148,17 @@ private void verifyUnmodifiedThrowingCheckedExceptions() throws CoderException {
       // encodedObject, because the Coder may treat it differently.
       //
       // For example, an unbounded Iterable will be encoded in an unbounded way, but decoded into an
-      // ArrayList, which will then be re-encoded in a bounded format. So we really do need to
-      // encode-decode-encode retainedObject.
-      if (Arrays.equals(
-          CoderUtils.encodeToByteArray(coder, clonedOriginalObject),
-          CoderUtils.encodeToByteArray(coder, clonedPossiblyModifiedObject))) {
-        return;
-      }
+      // ArrayList, which will then be re-encoded in a bounded format. So we get a structural value
+      // from a coder and used that to check if the byte array[] are the same. The structuralValue()
+      // method of the Coder returns a StructuralByteArray object.
```

Comment:
Hi @lukecwik Please I would like to have your point of view here, do not worry about additional changes I'll revert them